### PR TITLE
zigbee: Update Zigbee docs about configuring USB [KRKNWK-12057]

### PR DESCRIPTION
**Pull request by sebastiandraus**
- Update Zigbee documentation with latest changes in USB subsystem after zephyr upmerge
- Remove unused dependency in the Zigbee Light Switch sample
- Small documentation fixes

### CODE REVIEW

File: samples/zigbee/ncp/prj_usb.conf [hunk 1]
```diff
@@ -18,8 +18,7 @@ CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME="UART_1"
 # CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING=y
 # CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME="CDC_ACM_1"
 
-# Configure two USB CDC ACM instances for this sample: CDC_ACM_0 and CDC_ACM_1
-# CONFIG_USB_CDC_ACM_DEVICE_COUNT=2
+# Two USB CDC ACM instances cinfigured for this sample: CDC_ACM_0 and CDC_ACM_1
```
Comment (tomchy) [inline samples/zigbee/ncp/prj_usb.conf:21]: `cinfigured` -> `configured`

File: doc/nrf/libraries/zigbee/osif.rst [hunk 2]
```diff
@@ -114,6 +114,9 @@ Use the following options to configure the Zigbee serial logger:
   * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - This option specifies the serial device to use.
   * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE` - This option specifies the size of the internal ring buffer.
 
+  .. note::
+    See :ref:`zephyr:usb_device_cdc_acm` for more information of how to configure USB CDC ACM instance for logging ZBOSS trace messages.
```
Comment (tomchy) [inline doc/nrf/libraries/zigbee/osif.rst:118]: The preposition used in this sentence is most likely incorrect.

File: samples/zigbee/ncp/README.rst [hunk 3]
```diff
@@ -116,17 +116,33 @@ This can be configured by setting:
...
+      label = "CDC_ACM_0";
+   };
+
+   cdc_acm_uart1: cdc_acm_uart1 {
+      compatible = "zephyr,cdc-acm-uart";
+      label = "CDC_ACM_1";
+   };
+};
+
+If you have configured two USB CDC ACM instances, enable composite USB device driver using :kconfig:`CONFIG_USB_COMPOSITE_DEVICE`.
```
Comment (tomchy) [inline samples/zigbee/ncp/README.rst:138]: Does it mean that we have to update NCP sample configs?

**Review by tomchy — CHANGES_REQUESTED** [review]

File: samples/zigbee/ncp/README.rst (hunk 3 above)
Comment (sebastiandraus) [reply samples/zigbee/ncp/README.rst:138]: We don't. It's only used if second USB CDC ACM instance is configured to be used for logging binary ZBOSS traces over USB.

File: doc/nrf/libraries/zigbee/osif.rst (hunk 2 above)
Comment (sebastiandraus) [reply doc/nrf/libraries/zigbee/osif.rst]: Changed: `of` -> `about`

File: doc/nrf/libraries/zigbee/osif.rst [hunk 4]
```diff
@@ -114,6 +114,9 @@ Use the following options to configure the Zigbee serial logger:
   * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - This option specifies the serial device to use.
   * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE` - This option specifies the size of the internal ring buffer.
 
+  .. note::
+    See :ref:`zephyr:usb_device_cdc_acm` for more information about how to configure USB CDC ACM instance for logging ZBOSS trace messages.
```
Comment (greg-fer) [inline doc/nrf/libraries/zigbee/osif.rst:118]: Why not add this line or note next to line 113?
Also, please fix formatting of indent for lines 111-115 -- `*` should start from the beginning of the line.

File: doc/nrf/ug_zigbee_configuring.rst [hunk 5]
```diff
@@ -176,6 +176,8 @@ The stack logs are provided in a binary format and you can configure how they ar
       The application does not wait for the connection to be established and unreceived data is lost.
       For this reason, start collecting the data as soon as you need to.
 
+      See :ref:`zephyr:usb_device_cdc_acm` for more information about how to configure USB CDC ACM instance for logging ZBOSS trace messages.
+
```
Comment (greg-fer) [inline doc/nrf/ug_zigbee_configuring.rst:180]:
```suggestion
     The application does not wait for the connection to be established and unreceived data is lost.
     For this reason, start collecting the data as soon as you need to.

     See :ref:`zephyr:usb_device_cdc_acm` for more information about how to configure USB CDC ACM instance for logging ZBOSS trace messages.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: samples/zigbee/ncp/README.rst (hunk 3 above)
Comment (greg-fer) [inline samples/zigbee/ncp/README.rst:138]:
```suggestion
To create two instances of USB CDC ACM for application, complete the following steps:

1. Create two entries in the DTS overlay file for the selected board, one for each USB CDC ACM instance.
   See :ref:`zephyr:usb_device_cdc_acm` for more information.
#. While the NCP sample configures single USB CDC ACM instance ``"CDC_ACM_0"`` by default, configure the second USB CDC ACM instance ``"CDC_ACM_1"`` by extending ``zephyr_udc0`` node in the DTS overlay file with the following code:

   .. code-block:: console

      &zephyr_udc0 {
         cdc_acm_uart0: cdc_acm_uart0 {
            compatible = "zephyr,cdc-acm-uart";
            label = "CDC_ACM_0";
         };

         cdc_acm_uart1: cdc_acm_uart1 {
            compatible = "zephyr,cdc-acm-uart";
            label = "CDC_ACM_1";
         };
      };

#. Enable the composite USB device driver using the :kconfig:`CONFIG_USB_COMPOSITE_DEVICE` Kconfig option.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/zigbee/osif.rst [hunk 6]
```diff
@@ -108,11 +108,14 @@ Zigbee serial logger configuration options
...
 
-  * :kconfig:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - This option enables logging ZBOSS traces with Zigbee serial logger.
-  * :kconfig:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - This option selects the UART serial backend.
-  * :kconfig:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - This option selects the USB CDC ACM serial backend.
-  * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - This option specifies the serial device to use.
-  * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE` - This option specifies the size of the internal ring buffer.
+* :kconfig:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - This option enables logging ZBOSS traces with Zigbee serial logger.
+* :kconfig:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - This option selects the UART serial backend.
+* :kconfig:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - This option selects the USB CDC ACM serial backend.
+  .. note::
```
Comment (greg-fer) [inline doc/nrf/libraries/zigbee/osif.rst:114]: Blank line missing before the note.

File: samples/zigbee/ncp/README.rst [hunk 7]
```diff
@@ -116,17 +116,35 @@ This can be configured by setting:
...
 * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - to select the serial device for printing Zigbee stack logs, set to ``"CDC_ACM_1"``.
 
-To create two instances of USB CDC ACM for application, set the following:
+To create two instances of USB CDC ACM for application, complete the following steps:
 
-* :kconfig:`CONFIG_USB_CDC_ACM_DEVICE_COUNT` - defines number of USB CDC ACM devices to create, set to ``2`` in this configuration of NCP sample.
-* :kconfig:`CONFIG_USB_COMPOSITE_DEVICE` - to enable composite USB device driver.
+1. Create two entries in the DTS overlay file for the selected board, one for each USB CDC ACM instance.
+   See :ref:`zephyr:usb_device_cdc_acm` for more information.
+
```
Comment (greg-fer) [inline samples/zigbee/ncp/README.rst:123]: Blank line not needed.

File: samples/zigbee/ncp/README.rst [hunk 8]
```diff
@@ -116,17 +116,35 @@ This can be configured by setting:
...
+#. Enable the composite USB device driver using the :kconfig:`CONFIG_USB_COMPOSITE_DEVICE` Kconfig option.
 
 .. note::
     With this configuration, you will see two serial ports created by the NCP sample.
     Use the first one for NCP communication.
     Use second serial port for collecting Zigbee stack logs.
 
-Also, complete configuration of Zigbee stack logs over nRF USB serial is present in the :file:`prj_usb.conf`.
+Also, complete Kconfig configuration of Zigbee stack logs over nRF USB serial is present in the :file:`prj_usb.conf`.
```
Comment (greg-fer) [inline samples/zigbee/ncp/README.rst]:
```suggestion
You can also check the :file:`prj_usb.conf` file, which contains the complete Kconfig configuration of Zigbee stack logs over nRF USB serial commented out.
```

File: samples/zigbee/ncp/README.rst [hunk 9]
```diff
@@ -116,17 +116,35 @@ This can be configured by setting:
...
+            label = "CDC_ACM_1";
+         };
+      };
+
+#. Enable the composite USB device driver using the :kconfig:`CONFIG_USB_COMPOSITE_DEVICE` Kconfig option.
 
 .. note::
     With this configuration, you will see two serial ports created by the NCP sample.
     Use the first one for NCP communication.
     Use second serial port for collecting Zigbee stack logs.
```
Comment (greg-fer) [inline samples/zigbee/ncp/README.rst:144]:
```suggestion
With this configuration, you will see two serial ports created by the NCP sample.
Use the first one for NCP communication.
Use second serial port for collecting Zigbee stack logs.
```

File: samples/zigbee/ncp/README.rst [hunk 10]
```diff
@@ -116,17 +116,34 @@ This can be configured by setting:
 * :kconfig:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - to select nRF USB serial over UART serial.
 * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - to select the serial device for printing Zigbee stack logs, set to ``"CDC_ACM_1"``.
 
-To create two instances of USB CDC ACM for application, set the following:
+To create two instances of USB CDC ACM for application, complete the following steps:
```
Comment (greg-fer) [inline samples/zigbee/ncp/README.rst:119]: I'm being a bit picky here, but it's not clear to me why you need to create a second instance. The 117 line only suggests 1 of these 2 instances...

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: samples/zigbee/ncp/README.rst (hunk 10 above)
Comment (sebastiandraus) [reply samples/zigbee/ncp/README.rst:119]: One for NCP communication and one for ZBOO trace messages

**Review by greg-fer — APPROVED** [review]

**Review by maciejpietras — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

**Pull request merged**